apm-agent-plugins/apm-logging-plugin/apm-log4j2-plugin/src/main/java/co/elastic/apm/agent/log4j2/correlation/Log4j2LogCorrelationInstrumentation.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.log4j2.correlation;
+
+import co.elastic.apm.agent.bci.TracerAwareInstrumentation;
+import co.elastic.apm.agent.bci.bytebuddy.CustomElementMatchers;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.NamedElement;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+import java.security.ProtectionDomain;
+import java.util.Collection;
+import java.util.Collections;
+
+import static co.elastic.apm.agent.bci.bytebuddy.CustomElementMatchers.classLoaderCanLoadClass;
+import static co.elastic.apm.agent.bci.bytebuddy.CustomElementMatchers.implementationVersionGte;
+import static net.bytebuddy.matcher.ElementMatchers.hasSuperType;
+import static net.bytebuddy.matcher.ElementMatchers.isBootstrapClassLoader;
+import static net.bytebuddy.matcher.ElementMatchers.nameEndsWith;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.not;
+
+/**
+ * Instruments {@link org.apache.logging.log4j.core.impl.LogEventFactory#createEvent}
+ */
+public abstract class Log4j2LogCorrelationInstrumentation extends TracerAwareInstrumentation {
+
+    @Override
+    public Collection<String> getInstrumentationGroupNames() {
+        return Collections.singleton("log4j2-correlation");
+    }
+
+    @Override
+    public ElementMatcher.Junction<ClassLoader> getClassLoaderMatcher() {
+        return not(isBootstrapClassLoader())
+            .and(not(CustomElementMatchers.isAgentClassLoader()))
+            .and(classLoaderCanLoadClass("org.apache.logging.log4j.core.impl.LogEventFactor"));
```

Comment:
```suggestion
            .and(classLoaderCanLoadClass("org.apache.logging.log4j.core.impl.LogEventFactory"));
```

Reply:
Thanks, what a miss! 🤦‍♂️ 
Tests passing with this mean that `Log4j2_17_1InstrumentationTestRunner` and `Log4j2_6SInstrumentationTestRunner` are not being ran on CI. 
I am not sure if this is due to a mix between JUnit 4 and JUnit 5 on the same directory or due to test class naming. I'll look into that

Reply:
By default Maven Surefire does not consider `*TestRunner` a test, see https://maven.apache.org/surefire/maven-surefire-plugin/examples/inclusion-exclusion.html